apps/user/services/authentication.go

```diff
@@ -97,53 +97,104 @@ func (service AuthenticationService) BasicLogin(c *gin.Context) (
 
 // Identify ...
 func (service AuthenticationService) Identify(c *gin.Context) (
+	reSendTime int,
 	status int,
 	err error) {
 
 	var identifyData userSerializers.Identify
 	err = c.BindJSON(&identifyData)
 	if err != nil {
-		return http.StatusInternalServerError, err
+		return 0, http.StatusInternalServerError, err
 	}
 	gormDB := service.DB
 	var userData userModels.User
 	err = gormDB.Model(&userModels.User{}).Where("email = ?", identifyData.Email).Scan(&userData).Error
 	if err != nil {
 		if gorm.IsRecordNotFoundError(err) {
-			return http.StatusBadRequest, fmt.Errorf("We couldn't find a iReflect account associated with %s", identifyData.Email)
+			return 0, http.StatusBadRequest, fmt.Errorf("We couldn't find a iReflect account associated with %s", identifyData.Email)
 		}
-		return http.StatusInternalServerError, err
-	}
-	// when we don't need OTP to the email.
-	if !identifyData.SendOTP {
-		return http.StatusOK, nil
+		return 0, http.StatusInternalServerError, err
 	}
 	var otp userModels.OTP
 	err = gormDB.Model(&userModels.OTP{}).Where("user_id = ?", userData.ID).Scan(&otp).Error
-	if err != nil && !gorm.IsRecordNotFoundError(err) {
-		return http.StatusInternalServerError, err
+	if err != nil && err != gorm.ErrRecordNotFound {
```

Comment:
Why was it changed from the proper implementation?
```suggestion
	if err != nil && !gorm.IsRecordNotFoundError(err)  {
```

Reply:
done.